### DISCUSSION TITLE
form.isValid still true after form.setFieldError - bug or works as designed?

### DISCUSSION THREAD

**Discussion by filipjnc** [mantinedev/mantine#3029] [Q&A] [answered]
I have a form field for which I check if the value does not exist already in the database. 

I do this with a debounced callback on the `onChange` event. If the value exists in the db, I set a validation error on the field with `form.setFieldError("fieldName", "Value exists already")`. I want to disable the submit button when the validation has not succeeded. The `form.isValid()` returns true even after `form.setFieldError`. Is it a bug or expected behavior? 

What best practice is recommended for zod async validation?

**Comment by rtivital** [comment, marked as answer]
`use-form` is built with `useState` hooks, its state updates follow the same rules – you can get updated values only of the next render.

**Reply by filipjnc** [reply]
Thanks for the hint, I will try to mitigate it with the `useForceUpdate` hook.
